src/robot.coffee

```diff
@@ -211,6 +211,13 @@ class Robot
   loadFile: (path, file) ->
     ext  = Path.extname file
     full = Path.join path, Path.basename(file, ext)
+    if  require.cache[require.resolve(full)]
+      try
+        cacheobj = require.resolve(full)
+        @logger.debug "require cache for #{cacheobj} invalidated."
+        delete require.cache[cacheobj]
+      catch error
+        @logger.error "Unable to invalidate #{cacheobj}: #{error.stack}"
```

Comment:
Could we move this into the `try` block below instead? That way, it's only one try/catch block, it's logically grouped under the `if require.extensions[ext]`, and it also will use the hubot's error handling to report any issues with clearing the cache (ie `emit 'error'`).